pytest_motor/mongod_binary.py

```diff
@@ -0,0 +1,132 @@
+        # pylint: disable=consider-using-with
+        if str(self.url).endswith('.tgz'):
+            with tarfile.open(fileobj=binary_file) as archive:
+                file_in_archive = archive.extractfile(
+                    f'mongodb-{self.mongo_platform()}-{self.MONGO_VERSION}/bin/mongod')
+                file_outside = self.path.open(mode="wb")
+                with file_in_archive, file_outside:
+                    shutil.copyfileobj(file_in_archive, file_outside)
+                self.path.chmod(0o777)
+            assert self.path.exists(), "Unsuccessful mongod binary extraction"
+        elif str(self.url).endswith('.zip') and self.mongo_platform() == 'windows-x86_64':
+            with ZipFile(file=binary_file) as archive:
+                file_in_archive = archive.open(
+                    f'mongodb-win32-x86_64-windows-{self.MONGO_VERSION}/bin/mongod.exe')
+                file_outside = self.path.open(mode="wb")
+                with file_in_archive, file_outside:
+                    shutil.copyfileobj(file_in_archive, file_outside)
+                self.path.chmod(0o777)
+            assert self.path.exists(), "Unsuccessful mongod binary extraction"
+        # pylint: enable=consider-using-with
+        else:
+            raise Exception("Unsupported archive format.")  # pragma: no cover
+
+    @staticmethod
+    def mongo_running_os() -> str:
+        """Return OS as it spells in mongodb binary url."""
+        system: str = platform.system()
+        if system == 'Linux':
+            return 'linux'
+        if system == 'Darwin':
+            return 'osx'
+        if system == 'Windows':
+            return 'windows'
+        raise OSError("Your platform is unsupported.")
+
+    @staticmethod
+    def mongo_platform() -> str:
+        """Return mongo platform as it spells in mongodb binary url."""
+        # FUTURE: ARM support
+        system = platform.system()
+        if system == 'Linux':
+            distro_name = distro.id()
+            if distro_name == 'ubuntu':
+                return 'linux-x86_64-' + MongodBinary.__select_ubuntu_version()
+            if distro_name == 'debian':
+                return 'linux-x86_64-' + MongodBinary.__select_debian_version()
+            # FUTURE: distros that may be added: Amazon Linux, CentOS, SUSE
+            raise OSError(f"Your distro ({distro_name}) is unsupported.")
+
+        if system == 'Darwin':
+            return 'macos-x86_64'
+
+        if system == 'Windows':
+            return 'windows-x86_64'
+
+        raise OSError("Your platform is unsupported.")  # pragma: no cover
+
+    @staticmethod
+    def __select_ubuntu_version() -> str:
+        if bool(distro.major_version()) and int(distro.major_version()) < 16:
+            raise OSError("Your Ubuntu version is too old. Upgrade at least to 16.")
+
+        if distro.version() == '16.04':
+            warnings.warn("Your OS support was NOT properly tested.")
```

Comment:
Pull this into a helper method so that it can be called here and elsewhere?

Reply:
You mean:

```python
@staticmethod
def warn_untested_os() -> None:
    warnings.warn("Your OS support was NOT properly tested.")
```

like this?